### PR TITLE
Stop basing focused match color on matched colors

### LINKED ISSUE #5022: Search mode focused_match using incorrect CellBackground/CellForeground colors
### System

OS: Linux
Version: 0.7.2
Linux/BSD: X11, i3wm
Tested with the default config.
```
colors:
  search:
      matches:
        foreground: '#000000'
        background: '#ffffff'
      focused_match:
        foreground: CellBackground
        background: CellForeground
```
https://user-images.githubusercontent.com/33134232/116493381-798d7e00-a853-11eb-9a75-ed63188a5623.mp4

While in search mode, the focused match is expected to use the CellBackground/CellForeground colors like when selecting text. However, alacritty seems to be using the whatever colors are set under `search.matches` as the cell background/foreground.

### PR DESCRIPTION
**Pull request by nixpulvis**
Pending discussion and CHANGELOG entry.

fixes #5022

### CODE REVIEW

**Review by chrisduerr — CHANGES_REQUESTED** [review]
Fix comments, add changelog, change defaults of `focused_match` in yaml and code. Then it should be good to go.

**Comment by nixpulvis** [comment]
I really don't like the way this looks by default now however:

screenshot-2021-04-29T17:00:36Z

Trying to determine which of these is focused is quite hard.

**Comment by chrisduerr** [comment]
> I really don't like the way this looks by default now however:

As I've said, the defaults need to be changed they're currently incorrect. And if you're just talking about 'your defaults', that really doesn't matter.

**Comment by nixpulvis** [comment]
I'm thinking this might be a good default, with the matched text background in red, and the focused match inverted so-as to be consistent with selections:

Colored text focused (first `foo`):
screenshot-2021-04-29T17:23:23Z

Default foreground text focused (3rd `foo`):
screenshot-2021-04-29T17:23:33Z

**Comment by chrisduerr** [comment]
The focused match color should be the inverse of the normal text color.

**Comment by nixpulvis** [comment]
> The focused match color should be the inverse of the normal text color.

Isn't it? Or do you mean always like in the second image? If so, why? Isn't it nice that it's consistent with the way selections look in general?

**Comment by chrisduerr** [comment]
It's nice to keep things simple. There's no need to over-complicate this. The default config should behave the same way and the best way to do this is to just invert the normal match colors manually for the focused match.

File: alacritty.yml [hunk 1]
```diff
@@ -218,11 +218,11 @@
     # Allowed values are CellForeground/CellBackground, which reference the
     # affected cell, or hexadecimal colors like #ff00ff.
     #matches:
-    #  foreground: '#000000'
+    #  foreground: '#ff0000'
```
Comment (chrisduerr) [inline alacritty.yml:221]: I've told you that I don't want to change this, didn't I? I'm not sure how to make myself more clear on how I'd suggest to resolve this.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

**Comment by lockheadone** [comment]
https://github.com/alacritty/alacritty/pull/5025/files/70b42ab429dc103cb940809c530c0e04c38f3782

**Comment by nixpulvis** [comment]
> https://github.com/alacritty/alacritty/pull/5025/files/70b42ab429dc103cb940809c530c0e04c38f3782

Was there a point to this link?

File: alacritty.yml (hunk 1 above)
Comment (chrisduerr) [reply alacritty.yml:221]: There is zero reason to change the defaults in this PR. The requested change is completely unrelated to that. I'm not sure why you're derailing this PR with an unrelated change.

**Review by chrisduerr — APPROVED** [review]

**Pull request merged**